docs/reference/mapping/fields/synthetic-source.asciidoc

```diff
@@ -0,0 +1,161 @@
+[[synthetic-source]]
+==== Synthetic `_source`
+
+Though very handy to have around, the source field takes up a fair amount of
+space on disk. Instead of storing it on disk precisely as you sent it, Elasticsearch
+can reconstruct it on the fly. Enable this by setting `synthetic: true` in `_source`:
+
+[source,console,id=enable-synthetic-source-example]
+----
+PUT idx
+{
+  "mappings": {
+    "_source": {
+      "synthetic": true
+    }
+  }
+}
+----
+// TESTSETUP
+
+This on the fly reconstruction <<synthetic-source-modifications,modifies>> is *generally*
+slower than saving the source precisely and loading it, but it saves a lot of space.
+It also modifies the `_source` and is only supported if the index is entirely made up
+of the following field types:
+
+* <<boolean-synthetic-source,`boolean`>>
+* <<numeric-synthetic-source,`byte`>>
+* <<numeric-synthetic-source,`double`>>
+* <<numeric-synthetic-source,`float`>>
+* <<geo-point-synthetic-source,`geo_point`>>
+* <<numeric-synthetic-source,`half_float`>>
+* <<numeric-synthetic-source,`integer`>>
+* <<ip-synthetic-source,`ip`>>
+* <<keyword-synthetic-source,`keyword`>>
+* <<numeric-synthetic-source,`long`>>
+* <<numeric-synthetic-source,`scaled_float`>>
+* <<numeric-synthetic-source,`short`>>
+* <<text-synthetic-source,`text`>>
```

Comment:
```suggestion
While this on the fly reconstruction is *generally* slower than saving the source
documents verbatim and loading them at query time, it saves a lot of storage
space. There are a couple of other restrictions to be aware of:

* When you retrieve sythetic `_source` content it undergoes minor
<<synthetic-source-modifications,modifications>> compared to the original JSON.
* Synthetic `_source` can be used with indices that contain only these field
types:

** <<boolean-synthetic-source,`boolean`>>
** <<numeric-synthetic-source,`byte`>>
** <<numeric-synthetic-source,`double`>>
** <<numeric-synthetic-source,`float`>>
** <<geo-point-synthetic-source,`geo_point`>>
** <<numeric-synthetic-source,`half_float`>>
** <<numeric-synthetic-source,`integer`>>
** <<ip-synthetic-source,`ip`>>
** <<keyword-synthetic-source,`keyword`>>
** <<numeric-synthetic-source,`long`>>
** <<numeric-synthetic-source,`scaled_float`>>
** <<numeric-synthetic-source,`short`>>
** <<text-synthetic-source,`text`>>
```
I suggest breaking out the restrictions as separate bullets and rephrasing slightly.